typescript/ecs/capacity-providers-autoscaling/index.ts

```diff
@@ -0,0 +1,216 @@
+    },
+    {
+      capacityProvider: 'FARGATE',
+      base: 5, // corresponds min of count range
+      weight: 0,
+    },
+  ],
+});
+
+// Setup AutoScaling policy
+const scaling1 = service1.autoScaleTaskCount({
+  minCapacity: 5,
+  maxCapacity: 10,
+});
+scaling1.scaleOnCpuUtilization('CpuScaling', {
+  targetUtilizationPercent: 50,
+  scaleInCooldown: cdk.Duration.seconds(60),
+  scaleOutCooldown: cdk.Duration.seconds(60)
+});
+
+// Case 2: n services placed on spot instances, with scaling events triggering placement on spot only (i.e. 100% spot)
+const service2 = new ecs.FargateService(stack, 'FargateService-UserStory2a', {
+  cluster,
+  taskDefinition,
+  capacityProviderStrategies: [
+    {
+      capacityProvider: 'FARGATE_SPOT',
+      weight: 1,
+    },
+    {
+      capacityProvider: 'FARGATE',
+      weight: 0,
+    },
+  ],
+});
+
+// Setup AutoScaling policy
+const scaling2 = service2.autoScaleTaskCount({
+  minCapacity: 5,
+  maxCapacity: 10,
+});
+
+scaling2.scaleOnCpuUtilization('CpuScaling', {
+  targetUtilizationPercent: 50,
+  scaleInCooldown: cdk.Duration.seconds(60),
+  scaleOutCooldown: cdk.Duration.seconds(60)
+});
+
+/*
+ * Case 1bis: Up to p services placed on on-demand instances (where p >= n), with scaling events
+ * triggering placement on spot only.
+ */
+const service3 = new ecs.FargateService(stack, 'FargateService-DedicatedRange', {
+  cluster,
+  taskDefinition,
+  capacityProviderStrategies: [
+    {
+      capacityProvider: 'FARGATE_SPOT',
+      weight: 1,
+    },
+    {
+      capacityProvider: 'FARGATE',
+      base: 5,
+      weight: 0,
+    },
+  ],
+});
+
+const scaling3 = service3.autoScaleTaskCount({
+  minCapacity: 2,
+  maxCapacity: 10,
+});
+
+scaling3.scaleOnCpuUtilization('CpuScaling', {
+  targetUtilizationPercent: 50,
+  scaleInCooldown: cdk.Duration.seconds(60),
+  scaleOutCooldown: cdk.Duration.seconds(60)
+});
+
+/*
+ * Case 3: n services placed on on-demand instances, with scaling events triggering placement on spot or
+ * on-demand (mixed scaling). Here, scaling events would place 50% on spot and 50% on on-demand.
+ */
+const service4 = new ecs.FargateService(stack, 'FargateService-UserStory3', {
+  cluster,
+  taskDefinition,
+  capacityProviderStrategies: [
+    {
+      capacityProvider: 'FARGATE',
+      base: 5,
+      weight: 1,
+    },
+    {
+      capacityProvider: 'FARGATE_SPOT',
+      weight: 1,
+    },
+  ],
+});
+
+const scaling4 = service4.autoScaleTaskCount({
+  minCapacity: 5,
+  maxCapacity: 10,
+});
+
+scaling4.scaleOnCpuUtilization('CpuScaling', {
+  targetUtilizationPercent: 50,
+  scaleInCooldown: cdk.Duration.seconds(60),
+  scaleOutCooldown: cdk.Duration.seconds(60)
+});
+
+// Create ALB
+const lb = new elb.ApplicationLoadBalancer(stack, 'LB', {
+  vpc,
+  internetFacing: true
+});
+const listener = lb.addListener('PublicListener', { port: 80, open: true });
+
+// Attach ALB to ECS Service
+listener.addTargets('ECS', {
+  port: 80,
+  // never actually do this
```

Comment:
can we be more specific? I assume we mean "never actually take a load balancer and point it to 4 services' load balancers" Do we we really need line 179-208?